src/networking.c

```diff
@@ -2765,11 +2765,15 @@ static void setProtocolError(const char *errstr, client *c) {
         char buf[256];
         buf[0] = '\0';
         if (c->querybuf && sdslen(c->querybuf) - c->qb_pos < PROTO_DUMP_LEN) {
```

Comment:
Why not just do a check before getting into all this logic and print something like `Query buffer during protocol error <redacted>` right here and return? i.e. something like 

```
     if (server.hide_user_data_from_log) {
          snprintf(buf, sizeof(buf), "Query buffer during protocol error: <redacted>");
     } else {
         ....
     }
```

Or you can use a `goto` statement to simply jump to line 2786 below.